### PR TITLE
Move podsecuritypolicy registry to policy package

### PR DESCRIPTION
**Pull request by php-coder**
**What this PR does / why we need it:**
This is a part of the PSP migration from extensions to policy API group. This PR moves registry to policy package and changes preferred storage format to policy/v1beta1

**Which issue(s) this PR fixes:**
Addressed to https://github.com/kubernetes/features/issues/5

### CODE REVIEW

**Comment by php-coder** [comment]
/release-note-none
/sig auth

PTAL @liggitt @tallclair 
CC @simo5

File: pkg/registry/policy/podsecuritypolicy/storage/storage.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ import (
 	"k8s.io/kubernetes/pkg/printers"
 	printersinternal "k8s.io/kubernetes/pkg/printers/internalversion"
 	printerstorage "k8s.io/kubernetes/pkg/printers/storage"
-	"k8s.io/kubernetes/pkg/registry/extensions/podsecuritypolicy"
+	"k8s.io/kubernetes/pkg/registry/policy/podsecuritypolicy"
 )
 
 // REST implements a RESTStorage for PodSecurityPolicies.
```
Comment (liggitt) [inline pkg/registry/policy/podsecuritypolicy/storage/storage.go]: were we going to change `DefaultQualifiedResource: extensions.Resource("podsecuritypolicies"),` at this point?

File: cmd/kube-apiserver/app/server.go [hunk 2]
```diff
@@ -643,8 +643,7 @@ func BuildStorageFactory(s *options.ServerRunOptions, apiResourceConfig *servers
 	storageFactory.AddCohabitatingResources(apps.Resource("daemonsets"), extensions.Resource("daemonsets"))
 	storageFactory.AddCohabitatingResources(apps.Resource("replicasets"), extensions.Resource("replicasets"))
 	storageFactory.AddCohabitatingResources(api.Resource("events"), events.Resource("events"))
-	// TODO(#54933): 1.11: switch to using policy storage and flip the order here
-	storageFactory.AddCohabitatingResources(extensions.Resource("podsecuritypolicies"), policy.Resource("podsecuritypolicies"))
+	storageFactory.AddCohabitatingResources(policy.Resource("podsecuritypolicies"), extensions.Resource("podsecuritypolicies"))
```
Comment (liggitt) [inline cmd/kube-apiserver/app/server.go]: fix etcd_storage_path_test expected GVK

```
etcd_storage_path_test.go:663: GVK for PodSecurityPolicy from k8s.io/kubernetes/vendor/k8s.io/api/policy/v1beta1 does not match, expected extensions/v1beta1, Kind=PodSecurityPolicy got policy/v1beta1, Kind=PodSecurityPolicy
etcd_storage_path_test.go:663: GVK for PodSecurityPolicy from k8s.io/kubernetes/vendor/k8s.io/api/extensions/v1beta1 does not match, expected extensions/v1beta1, Kind=PodSecurityPolicy got policy/v1beta1, Kind=PodSecurityPolicy

```

File: pkg/registry/policy/podsecuritypolicy/storage/storage.go (hunk 1 above)
Comment (php-coder) [reply pkg/registry/policy/podsecuritypolicy/storage/storage.go]: Do you know what `DefaultQualifiedResource` does? What role it plays and what will be impact of this change? Last time I changed it, I introduced strange issue that only @sttts was able to debug.

**Comment by php-coder** [comment]
/assign @liggitt @sttts 

PTAL

**Comment by liggitt** [comment]
/lgtm
thanks

**Pull request merged**